aptos-move/framework/supra-framework/sources/transaction_validation.move

```diff
@@ -167,6 +167,35 @@ module supra_framework::transaction_validation {
         )
     }
 
+    fun automated_tranaction_prologue(
+        sender: signer,
+        txn_gas_price: u64,
+        txn_max_gas_units: u64,
+        txn_expiration_time: u64,
+        _script_hash: vector<u8>,
```

Comment:
In the old prologue functions, I noticed that we pass chain_id and validate it using an assertion, such as `assert!(chain_id::get() == chain_id, error::invalid_argument(PROLOGUE_EBAD_CHAIN_ID))`.
Do we still need this validation in the `automated_transaction_prologue`?

BTW, there is a typo in `automated_transaction_prologue` — `tranaction`.

Reply:
Thanks for the hint. also _script_hash should be removed from here.
as for chain-id, it can be there or not.
`AutomatedTransaction` will be constructed internally by `smr-moonshot-executor` component, and the chain ID will be set there. so theoretically it should be always valid.
But as long as `smr-moonshot-executor` and `MoveVM`  are completly isolated components, and one provides input to the other, in my understanding, each component should validate its input and not rely on the fact that the other one will not produce any buggy information. This way the if there is a bug in one layer which is not tested, then the other layer can at least point it out.
But we need to agree whether we want to go with this approach or not.